src/common/verbose.cpp

```diff
@@ -15,7 +15,6 @@
 *******************************************************************************/
 
 #include <atomic>
-#include <iostream>
```

Comment:
Can you please elaborate on the impact from having global-static? How important is it for you?
We don't monitor such places hence we cannot guarantee that it won't happen in the future.

Reply:
It adds code that gets run at load time. That's more code than no code.

It impacts us because we need to make sure our tool has no load-time code that runs before we perform a check that the CPU is new enough. We're considering using oneDNN, but we need to remove all load-time code first.

Reply:
So having the load-time code can interfere with results of the check? How critical for the check would be creating a global variable?

In general, moving global variables to a function scope to initialize them on demand is the right thing to do, but as I said we don't test such things therefore even if all the places are fixed now we can't guarantee that it won't happen gain.

Reply:
No, it can interfere with the application launching. We need to be sure that the application is being run on a CPU that has all the CPU features that oneDNN and all the other content was compiled for. I much prefer to receive a message saying:

```
Cannot run on this CPU.
This application requires certain features not found in your CPU:
avx,f16c,avx2,fma
```

Than a crash with SIGILL.

Reply:
This seems brittle given that a simple `iostream` header can become a problem. Why don't load oneDNN at runtime and use C interfaces to perform the check?

Reply:
Yes, it's brittle. But it's a choice libstdc++ maintainers made back in 2004 (GCC 3.4 times), possibly earlier. It can't be fixed without breaking the ABI, AFAIK.

We can't load oneDNN at runtime because we must deliver a fully static executable to one of our customers (a very important one for you and I). Dynamic loading is not possible.

Reply:
I see, thanks for the explanation.